### PR TITLE
feat: Unified Error handling

### PR DESCRIPTION
**Pull request by inderps**
Screenshot 2024-07-04 at 10 54 42
Screenshot 2024-07-04 at 11 10 13
Screenshot 2024-07-04 at 11 10 20
Screenshot 2024-07-04 at 11 23 27
Screenshot 2024-07-04 at 12 02 02

### CODE REVIEW

**Review by zalborzi — DISMISSED** [review]

File: src/app/DatasetCounter.tsx [hunk 1]
```diff
@@ -2,20 +2,35 @@
...
-      setCount(count);
+      try {
+        const count = await datasetCount();
+        setCount(count);
+      } catch (error) {
+        if (error instanceof AxiosError) {
+          setAlert({
+            type: "error",
+            message:
+              error.response?.data?.title ||
```
Comment (brunopacheco1) [inline src/app/DatasetCounter.tsx:24]: suggestion: there is no need to add fallback for `title`, as we are implementing [RFC 7807](https://datatracker.ietf.org/doc/html/rfc7807).

`detail` is not mandatory, by this spec, but in our APIs we always have details.

`status` is mandatory and is always there as well.

File: src/providers/application/ApplicationProvider.tsx [hunk 2]
```diff
@@ -349,4 +341,19 @@ function useApplicationDetails() {
   return context;
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const handleErrors = (error: any, fallbackMessage: string, dispatch: any) => {
+  if (error instanceof AxiosError) {
+    dispatch({
+      type: ApplicationActionType.REJECTED,
+      payload: `Status: ${error.response?.status} - ${error.response?.data}`,
```
Comment (brunopacheco1) [inline src/providers/application/ApplicationProvider.tsx:349]: question: is this going to be show a json to the UI alert?

File: src/providers/application/ApplicationProvider.tsx [hunk 3]
```diff
@@ -202,10 +203,7 @@ function ApplicationProvider({ children }: ApplicationProviderProps) {
       const response = await saveFormAndDuos(updatedForms);
       if (response.ok) fetchApplication();
     } catch (error) {
-      dispatch({
-        type: ApplicationActionType.REJECTED,
-        payload: "Failed to add attachment",
-      });
+      handleErrors(error, "Failed to add attachment", dispatch);
```
Comment (brunopacheco1) [inline src/providers/application/ApplicationProvider.tsx:206]: suggestion: there is no need for fallback messages.

**Review by brunopacheco1 — COMMENTED** [review]
Can we have a look at this PR together?

File: src/app/DatasetCounter.tsx (hunk 1 above)
Comment (brunopacheco1) [reply src/app/DatasetCounter.tsx:24]: As agreed, it makes sense to have fallback title, if the exception is coming from another layer - e.g. Kong - and that layer does not implement this RFC.

**Review by brunopacheco1 — APPROVED** [review]

**Pull request merged**